ocrd_segment/evaluate_address.py

```diff
@@ -0,0 +1,285 @@
+                    'char-error-rate-mean': cedits[i].mean,
+                    'char-error-rate-varia': cedits[i].varia,
+                    'word-error-rate-mean': wedits[i].mean,
+                    'word-error-rate-varia': wedits[i].varia,
+                }
+            if confusion:
+                for i in range(1, len(ifgs)):
+                    if not cedits[i].length:
+                        continue
+                    conf = self.caligners[i].get_confusion(confusion)
+                    LOG.info("most frequent confusion / %s vs %s: %s",
+                            ifgs[0], ifgs[i], conf)
+                    report[ifgs[0] + ',' + ifgs[i]]['confusion'] = repr(conf)
+            if histogram:
+                for i in range(1, len(ifgs)):
+                    if not cedits[i].length:
+                        continue
+                    hist = cedits[i].hist()
+                    LOG.info("character histograms / %s vs %s: %s",
+                            ifgs[0], ifgs[i], hist)
+                    report[ifgs[0] + ',' + ifgs[i]]['histogram'] = repr(hist)
+            # write back result to overall report
+            file_id = self.output_file_grp
+            file_path = os.path.join(self.output_file_grp, file_id + '.json')
+            self.workspace.add_file(
+                ID=file_id,
+                file_grp=self.output_file_grp,
+                pageId=None,
+                local_filename=file_path,
+                mimetype='application/json',
+                content=json.dumps(report, indent=2, ensure_ascii=False))
+
+
+def page_get_lines(pcgts):
+    '''Get all TextLines in the page.
+    
+    Iterate the element hierarchy of the page `pcgts` down
+    to the TextLine level. For each line, store the element
+    ID and its first TextEquiv annotation.
+    
+    Return the stored dictionary.
+    '''
+    LOG = getLogger('processor.EvaluateLines')
+    result = dict()
+    regions = pcgts.get_Page().get_AllRegions(classes=['Text'], order='reading-order')
+    if not regions:
+        LOG.warning("Page contains no text regions")
+    for region in regions:
+        LOG.info("GET ADDRESS TYPE: %s", str(get_address_type(region)))
+        address_type = get_address_type(region)
+        if address_type in CATEGORIES:
+            lines = region.get_TextLine()
+            if not lines:
+                LOG.warning("Region '%s' contains no text lines", region.id)
+                continue
+            for line in lines:
+                textequivs = line.get_TextEquiv()
+                if not textequivs:
+                    LOG.warning("Line '%s' contains no text results", line.id)
+                    continue
+                LOG.info("textequivs in line %s with text %s", str(line.id), str(textequivs[0].Unicode))
+                result[line.id] = textequivs[0].Unicode
+                result[CATEGORIES[CATEGORIES.index(address_type)]] = textequivs[0].Unicode
```

Comment:
I don't understand that formulation. How is this different from just `result[address_type]`?

Also, what worth does it have to get the CER between
- two address-rcpt instances
- two address-sndr instances
- two address-contact instances
- two ADDRESS_STREET_HOUSENUMBER_ZIP_CITY instances
- two ADDRESS_ZIP_CITY instances
- two ADDRESS_FULL instances

if they are not from the same physical location (because one is misclassified or because there are multiple different instances)? (I thought we'd go the other way: first identify overlaps and matches/splits/merges, then align, then calculate CER/distance – with and without classification...)